EventsExpress/ClientApp/src/containers/event-filter.js

```diff
@@ -1,98 +1,97 @@
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
-import {  getFormValues, reset } from 'redux-form';
+import { getFormValues, reset } from 'redux-form';
 import EventFilter from '../components/event/event-filter';
-import { get_events,get_eventsForAdmin } from '../actions/event-list';
-import history from '../history';
-
+import { updateEventsFilters } from '../actions/event-list';
 import get_categories from '../actions/category-list';
+import eventHelper from '../components/helpers/eventHelper';
 
 class EventFilterWrapper extends Component {
-
-    componentWillMount(){
+    componentWillMount() {
         this.props.get_categories();
     }
 
     onReset = () => {
         this.props.reset_filters();
-        var search_string = '?page=1';
-        if(window.location.search !== search_string){
-        if(this.props.current_user.role==="Admin"){
-            this.props.AdminSearch(search_string); 
-        }
-        else{
-            this.props.search(search_string); 
-        }
-        history.push(window.location.pathname + search_string);   
+        this.props.updateEventsFilters(eventHelper.getDefaultEventFilter());
     }
+
+    onLoadUserDefaults = () => {
+        this.props.reset_filters();
+        const defaultFilter = {
+            ...eventHelper.getDefaultEventFilter(),
+            categories: this.props.current_user.categories.map(item => item.id),
+        };
+
+        this.props.updateEventsFilters(defaultFilter);
     }
-    
-    onSubmit = (filters) => {  
 
-        var search_string = '?page=1';
-        if (filters != null) {
-            if (filters.search != null) {
-                search_string += '&keyWord=' + filters.search;
-            }
-            if (filters.dateFrom != null) {
-                search_string += '&dateFrom=' + new Date(filters.dateFrom).toDateString();
-            }
-            if (filters.dateTo != null) {
-                search_string += '&dateTo=' + new Date(filters.dateTo).toDateString();
-            }
-            if (filters.categories != null) {
-                var categories = '';
-                for (var i = 0; i < filters.categories.length; i++) {
-                    categories += filters.categories[i].id + ',';
-                }
-                search_string += '&categories=' + categories;
-            }
-            if(filters.status==="all"){
-                search_string+='&All='+true;
+    onSubmit = (filters) => {
+        filters = JSON.parse(JSON.stringify(filters));
```

Comment:
Why is this required?

Reply:
This is required to trim keys with undefined values. This code has been refactored.